Tests/JaegerTests/JaegerJSONSenderTests.swift

```diff
@@ -0,0 +1,131 @@
+//
+//  Created by Simon-Pierre Roy on 12/20/18.
+//
+
+import XCTest
+@testable import Jaeger
+
+class JaegerJSONSenderTests: XCTestCase {
+
+    func testSendSpansNoError() {
+
+        let urlTest = URL(string: "testURL")!
+        let mockSession = URLSessionMock()
+        let testProcess = JaegerBatchProcess(serviceName: "test", tags: [])
+        let sender = JaegerJSONSender(endPoint: urlTest, process: testProcess, session: mockSession)
+
+        let spansSent = XCTestExpectation(description: "sender sent spans")
+
+        let jaegerSpan = JaegerSpan(span: TestUtilities.getNewTestSpan())
+
+        sender.send(spans: [jaegerSpan]) { error in
+            XCTAssertNil(error)
+            spansSent.fulfill()
+        }
+
+        wait(for: [spansSent], timeout: 1)
+    }
+
+    func testSendSpansWithError() {
+
+        let urlTest = URL(string: "testURL")!
+        let mockSession = URLSessionMock()
+        let testError = NSError(domain: "testError", code: -1, userInfo: nil)
+
+        mockSession.error = testError
+
+        let testProcess = JaegerBatchProcess(serviceName: "test", tags: [])
+        let sender = JaegerJSONSender(endPoint: urlTest, process: testProcess, session: mockSession)
+
+        let spansSent = XCTestExpectation(description: "sender sent spans")
+        let jaegerSpan = JaegerSpan(span: TestUtilities.getNewTestSpan())
+
+        sender.send(spans: [jaegerSpan]) { error in
+            XCTAssertEqual((error as NSError?), testError)
+            spansSent.fulfill()
+        }
+
+        wait(for: [spansSent], timeout: 1)
+    }
+
+    func testSendSpansJaeger() {
+
+        let urlTest = URL(string: "testURL")!
+        let mockSession = URLSessionMock()
+
+        let nameId = "test process"
+
+        let tag = Tag(key: nameId, tagType: .bool(true))
+        let testProcess = JaegerBatchProcess(serviceName: nameId, tags: [tag])
+        let sender = JaegerJSONSender(endPoint: urlTest, process: testProcess, session: mockSession)
+
+        let spansSent = XCTestExpectation(description: "sender sent spans")
+        let sessionDone = XCTestExpectation(description: "session Done")
+        let jaegerSpan = JaegerSpan(span: TestUtilities.getNewTestSpan(name: nameId ))
+
+        mockSession.dataTaskExcuted = { resquest in
+            guard let data = resquest.httpBody else { return XCTFail("Need Data") }
+            guard let batch = try? JSONDecoder().decode(JaegerBatch.self, from: data)  else {
+                return XCTFail("Wrong Data")
+            }
+            XCTAssertEqual(batch.process.serviceName, nameId)
+            XCTAssertEqual(batch.process.tags?.first?.key, nameId)
+            XCTAssertEqual(batch.spans.first?.operationName, nameId)
+            spansSent.fulfill()
+        }
+
+        sender.send(spans: [jaegerSpan]) { _ in
+            sessionDone.fulfill()
+        }
+
+        wait(for: [spansSent, sessionDone], timeout: 1)
+    }
+
+    func testSendSpansJaegerNotJaegerSpans() {
```

Comment:
Function name is a little unclear. Are we testing the behavior of sending a non-jaeger span using a Jaeger JSON sender?

Reply:
Yes! Do you have a good name?

Reply:
testJaegerJSONSenderWithNonJaegerSpans